src/write.c

```diff
@@ -376,34 +370,43 @@ static void avifItemPropertyDedupStart(avifItemPropertyDedup * dedup)
 
 // This compares the newly written item property (in the dedup's temporary storage buffer) to
 // already-written properties (whose offsets/sizes in outputStream are recorded in the dedup). If a
-// match is found, the previous item's index is used. If this new property is unique, it is
+// match is found, the previous property index is used. If this new property is unique, it is
 // assigned the next available property index, written to the output stream, and its offset/size in
 // the output stream is recorded in the dedup for future comparisons.
 //
-// This function always returns a valid 1-indexed property index for usage in a property association
-// (ipma) box later. If the most recent property was a duplicate of a previous property, the return
-// value will be the index of the original property, otherwise it will be the index of the newly
+// On success, this function adds to the given ipma box the association linking the reused or newly
```

Comment:
Nit/optional: "the association" => "an association" or "a property association" ?

Also, the sentence sounds incomplete because it leaves me wondering "linking the property with what?" Perhaps add "to the item" or "with the item" at the end of the sentence?